Apromore-Core-Components/Apromore-Portal/src/main/java/org/apromore/portal/security/helper/JwtHelper.java

```diff
@@ -0,0 +1,187 @@
+            }
+        }
+
+        return null;
+    }
+
+    public static boolean isJwtExpired(final JWTClaimsSet jwtClaimsSet,
+                                       final String issuedAtStr,
+                                       final String expiredAtStr) {
+        long jwtExpiryEpoch;
+        if (expiredAtStr != null) {
+            jwtExpiryEpoch = Long.valueOf(expiredAtStr.substring(3));
+        } else { // Calculate relative
+            final Long jwtIssuedAt = Long.valueOf(issuedAtStr.substring(3));
+
+            jwtExpiryEpoch = calculateExpiryTime(jwtIssuedAt, WEBAPP_SSO_SESSION_TIMEOUT);
+        }
+
+        final Long nowEpoch = LocalDateTime.now().atZone(ZoneId.systemDefault()).toEpochSecond();
+
+        final boolean jwtExpired = (nowEpoch > jwtExpiryEpoch);
+
+        return jwtExpired;
+    }
+
+    public static JWTClaimsSet getClaimsSetFromJWT(final String jwtStr) throws Exception {
+        final JWT jwtParsed = JWTParser.parse(jwtStr);
+        final JWTClaimsSet jwtClaimsSet = jwtParsed.getJWTClaimsSet();
+
+        return jwtClaimsSet;
+    }
+
+    /**
+     * Return authenticated user if JWT token is not expired, otherwise <code>null</code>.
+     *
+     * @return
+     */
+    public static UserType userFromJwt(final ManagerService managerService, final String appAuthHeader) {
+        try {
+            final JWTClaimsSet jwtClaimsSet = JwtHelper.getClaimsSetFromJWT(appAuthHeader);
+
+            final String issuedAtStr = jwtClaimsSet.getStringClaim(JwtHelper.STR_JWT_KEY_ISSUED_AT);
+            final String expiryAtStr = jwtClaimsSet.getStringClaim(JwtHelper.STR_JWT_EXPIRY_TIME);
+
+            final boolean jwtExpired = JwtHelper.isJwtExpired(jwtClaimsSet, issuedAtStr, expiryAtStr);
+
+            if (!jwtExpired) {
+                final UserType authUserType =
+                        buildAuthenticatedUserFromJwt(jwtClaimsSet, managerService);
+
+                return authUserType;
+            } else {
+                return null;
+            }
+        } catch (final Exception pe) {
+            return null;
+        }
+    }
+
+    public static UserType buildAuthenticatedUserFromJwt(final JWTClaimsSet jwtClaimsSet,
+                                                         final ManagerService managerService) {
+        final String username = (String)jwtClaimsSet.getClaim(JwtHelper.JWT_KEY_SUBJECT_USERNAME);
+
+        final UserType userType = managerService.readUserByUsername(username);
+
+        return userType;
+    }
+
+    public static User convertFromUserType(final UserType userType, final SecurityService securityService) {
+        final DateFormat formatter = new SimpleDateFormat("dd/MM/yyyy");
+        Date date = null;
+        if (userType.getLastActivityDate() != null && !userType.getLastActivityDate().equals("")) {
+            try {
+                date = formatter.parse(userType.getLastActivityDate());
+            } catch (ParseException ex) {
+            }
```

Comment:
If it's really okay for there to be unparseable dates that pass through as `null`, I'd be inclined to at least have a comment confirming that this is deliberate.  Logging the unparseable date as a warning wouldn't go astray, either.

Reply:
Added logged comment (done).